server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -110,10 +111,14 @@
     private final Map<TcpChannel, ChannelPendingTaskTracker> channelPendingTaskTrackers = ConcurrentCollections.newConcurrentMap();
     private final SetOnce<TaskCancellationService> cancellationService = new SetOnce<>();
 
+    /** Consumers that are notified of the stats */
+    private List<TaskStatConsumer> statConsumers;
```

Comment:
I believe we should use the thread-safe data structure here, `TaskManager` is a singleton and `addTaskStatConsumer` could be called any time. Alternatively, we could make it `final` and add the constructor argument (instead of `addTaskStatConsumer`) to exclude any races.